examples/applyconfig/main.go

```diff
@@ -0,0 +1,141 @@
+	// The types defined must still be registered in the normal way
+	bld := &scheme.Builder{GroupVersion: schema.GroupVersion{
+		Group:   "applytest.kubebuilder.io",
+		Version: "v1",
+	}}
+	bld.Register(&v1.Foo{})
+	bld.AddToScheme(sch)
+
+	cl, err := client.New(cfg, client.Options{Scheme: sch})
+	if err != nil {
+		panic(err)
+	}
+
+	// Create two owners for two apply clients
+	var owner client.FieldOwner
+	owner = "fieldmanager_test"
+	var owner2 client.FieldOwner
+	owner2 = "fieldmanager_test2"
+
+	// ====================================================
+	// Without ApplyConfigurations
+	// Please skip to the next section to see a working example with ApplyConfigurations
+	// This example is used to show the limitations of the current structs
+	// ====================================================
+
+	apply1 := &v1.Foo{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Foo",
+			APIVersion: "applytest.kubebuilder.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "normalobject",
+			Namespace: "default",
+		},
+		NonNullableField: "test",
+	}
+
+	// Suppose the first apply is executed and we want to modify a field not owned by the first apply (NullableField)
+	// This apply will fail because NonNullableField is defaulted to "" and the second applier will attempt to own
+	// the NonNullableField despite leaving it empty. Since the first applier already owns the field, the apply will fail
+	apply2 := &v1.Foo{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Foo",
+			APIVersion: "applytest.kubebuilder.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "normalobject",
+			Namespace: "default",
+		},
+		NullableField: "value",
+	}
+
+	if err := cl.Patch(context.TODO(), apply1, client.Apply, owner); err != nil {
+		panic(err)
+	}
+
+	if err := cl.Patch(context.TODO(), apply2, client.Apply, owner2); err == nil {
+		fmt.Errorf("Expected a conflict but no conflict was hit")
+	}
+
+	// ====================================================
+	// Using ApplyConfigurations Structs
+	// ====================================================
+
+	fooObj := &v1.Foo{}
+
+	// To use the SSA typed client, use the ApplyConfiguration generated types instead of the types defined in v1
+	applyConfig1 := (&ac.FooApplyConfiguration{
+		TypeMetaApplyConfiguration:   *metav1ac.TypeMeta().WithKind("Foo").WithAPIVersion("applytest.kubebuilder.io/v1"),
+		ObjectMetaApplyConfiguration: metav1ac.ObjectMeta().WithName("acdefault").WithNamespace("default"),
+	}).WithNonNullableField("value1")
+
+	// Without ApplyConfiguration, NonNullableField is not an optional field so it must be specified if Foo{} was used instead of the ApplyConfiguration
+	// The ApplyConfiguration uses pointers for all objects, so omitting a field is equivalent to saying that it is not of interest in this Apply
+	applyConfig2 := (&ac.FooApplyConfiguration{
+		TypeMetaApplyConfiguration: *metav1ac.TypeMeta().WithKind("Foo").WithAPIVersion("applytest.kubebuilder.io/v1"),
```

Comment:
Do we need to set type meta? We can derive this from the `obj` in patch or not?

Reply:
can you point me in the right direction for finding this? I wasn't find an easy way to derive this since the applyconfig objects are constructed rather than obtained from the apiserver.